### ISSUE TITLE
SPI Review of GitHubService.create

### ISSUE THREAD

**Issue by ALRubinger** [redhat-kontinuity/catapult#58]
Currently:

```
GitHubRepository create(String repositoryName, String description, String homepage,
        boolean has_issues, boolean has_wiki, boolean has_downloads) throws IOException, IllegalArgumentException
```

...the impl has:

```
@Override
    public GitHubRepository createRepository(String repositoryName,
                                   String description,
                                   String homepage,
                                   boolean hasIssues,
                                   boolean hasWiki,
                                   boolean hasDownloads) throws IOException, IllegalArgumentException
```

Let's:

1) Drop the `IOException` from the SPI and on the impl side wrap it in a RuntimeException
2) Ensure there's test coverage in `GitHubServiceIT`
3) Drop the "has" issues, wiki, and downloads from what we provide (we'll set these flags to false on the impl side).  The user can always add these if they want from the GitHub Web UI, but we won't ever be prompting them for this info if we make new repos on their behalf.
4) Drop `homepage`

Basically let's get the SPI to read:

```
/**
     * Creates a repository with the given information.  Issues, Wiki, and Downloads will be set to disabled.
     *
     * @param repositoryName - the name of the repository
     * @param description - the repository description
     * @return the created {@link GitHubRepository}
     * @throws IllegalArgumentException
     */
GitHubRepository createRepository(String repositoryName, String description) throws IllegalArgumentException
```

**Comment by xcoulon** [comment]
Should this method be moved in the 'GitHubServiceSpi' interface, as mentionned in https://github.com/redhat-kontinuity/catapult/issues/61 ? Or is there a use case for creating a repository (other than in tests) ?

**Comment by xcoulon** [comment]
In which case, the GitHubResource#callbackString, String) method should be simplified, too.

**Comment by ALRubinger** [comment]
Yes, this is now in the SPI.

Eventually we may have a use case for creating a repo, but until then, let's hide it from the user API.  Can always move it later when it matures.

Sent @xcoulon an invite to the org; when he accepts I can assign this to him.

**Comment by rimolive** [comment]
Please review e179269 so I can push the fix for issue #63

**Comment by ALRubinger** [comment]
7fdba8f5fc2373e93cd2c3bdabfcd7451f0cc635
